### PR TITLE
fix(debian/control): get working with `crossbuilder`

### PR DESCRIPTION
**Pull request by myii**
The main discussion behind this PR took place in the `UBports Core App Developers` Matrix room.  Bringing the main bits here as a draft PR at first, in order to discuss how to proceed.

> OK, to get to the bottom of this, I prepared everything on another computer, so that I'd be forced to do all of the stages again. I've figured out the missing stage (or two):
> 
> So with the default settings, this is what you're faced with:
> 
> ```bash
> The following packages have unmet dependencies:
>  builddeps:ubuntu-system-settings:armhf : Depends: dh-migrations:armhf but it is not installable
>                                           Depends: qtbase5-private-dev:armhf but it is not going to be installed
>                                           Depends: pep8:armhf but it is not installable
>                                           Depends: python3-pep8:armhf but it is not installable
>                                           Depends: python3-xdg:armhf but it is not installable
> ```
> 
> Do the following to deal with all but one of these:
> 
> * (a) `crossbuilder inst-foreign dh-migrations`
> * (b) Use the diff shown above in `debian/control`
> 
> That still leaves:
> 
> ```bash
>  builddeps:ubuntu-system-settings:armhf : Depends: qtbase5-private-dev:armhf but it is not going to be installed
> ```
> 
> For this, you need to enter the LXD container and install it there:
> 
> ```bash
> $ crossbuilder shell
> # sudo apt install qtbase5-private-dev:armhf
> ```
> 
> * Exit the container's shell as usual.
> 
> Then everything should work fine with:
> 
> ```bash
> $ crossbuilder --no-deb
> ```

* This PR is the diff mentioned in `(b)` above.

Going forward, Jonny suggested that this process could be added to the README.  I haven't done that so far since I'd rather it was agreed what the correct process is.

### CODE REVIEW

**Comment by jonnius** [comment]
@mardy Could you have a look at this? Especially why it is necessary to enter the lxd and install `sudo apt install qtbase5-private-dev:armhf`?

File: debian/control [hunk 1]
```diff
@@ -102,7 +102,7 @@ Depends: ${misc:Depends},
          python3,
          python3-gi,
          python3-dbus,
-         python3-xdg,
+         python3-xdg <!nocheck>,
```
Comment (dobey) [inline debian/control:105]: This is wrong. The `<!nocheck>` is only applicable to build dependencies.

File: debian/control [hunk 2]
```diff
@@ -51,13 +51,13 @@ Build-Depends: dbus-test-runner,
                libqt5sql5-sqlite,
                xvfb,
                cmake,
-               pep8,
-               python3-pep8,
-               python3-flake8 (>= 2.2.2-1ubuntu4) | python3-flake8:native,
+               pep8 <!nocheck>,
+               python3-pep8 <!nocheck>,
+               python3-flake8:native (>= 2.2.2-1ubuntu4) | python3-flake8:native,
```
Comment (dobey) [inline debian/control:56]: Hrmm. Both sides of this or dependency are the same, and I'm not sure flake8 is used anyway, so this dep should be removed. Instead perhaps having `python3-pyflakes <!nocheck>` would be better (as it's what is being used by the tests).

File: debian/control [hunk 3]
```diff
@@ -6,7 +6,7 @@ Build-Depends: dbus-test-runner,
                debhelper (>= 9),
                dh-migrations,
                dh-python,
-               python3-setuptools,
+               python3-setuptools:native,
```
Comment (dobey) [inline debian/control:9]: This is an `all` architecture package, so the `:native` here is meaningless.

**Review by dobey — CHANGES_REQUESTED** [review]

**Comment by dobey** [comment]
> @mardy Could you have a look at this? Especially why it is necessary to enter the lxd and install `sudo apt install qtbase5-private-dev:armhf`?

This is probably due to a conflict with another dependency as system-settings isn't quite right for the crossbuilder scenario yet.

**Comment by myii** [comment]
@dobey Thanks for the review.  I'll incorporate those changes as soon as I get a chance.

**Comment by mardy** [comment]
Hi @myii, and thanks for working on this! I worked on the same issue a few months ago as part of some other development story, and I had a pretty decent patch here: https://github.com/ubports/system-settings/pull/84/commits/0e493fdda0500a6343a39de8498dd1489bedc744

Tonight I'll check if it still works and, if so, I'll push it for review.

**Comment by myii** [comment]
@mardy Thanks for that feedback.  I hope your patch works because I've completely messed up my `crossbuilder` dependencies in testing I've done since.  Each package blocking the other -- dependency hell!  At least it's only on a test machine and not on one that I use for real.

**Comment by UniversalSuperBox** [comment]
Superseded by https://github.com/ubports/system-settings/pull/148

**Pull request closed without merging**